.github/workflows/docker-publish-workflow.yml

```diff
@@ -0,0 +1,50 @@
+# Whenever a new release is published, this action will build a new Docker image
+# and push it to DockerHub. This action is based on the following documentation:
+# https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
+# https://github.com/docker/build-push-action/blob/master/docs/advanced/dockerhub-desc.md
+
+name: Publish Docker image
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: cqframework/cql-translation-service
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
```

Comment:
This is the only thing I want to just double check on. This is saying build and push everything in the current directory, right?

Reply:
It's saying the _context_ of the build is the root directory.  This is basically telling it where to find the `Dockerfile` and what current directory to use when running the docker build command.  It doesn't necessarily build or push _everything_ in the folder -- the Dockerfile controls what files/paths are used in the build.

Reply:
See https://github.com/docker/build-push-action#inputs for description of the arguments.